### PR TITLE
Included support for GLP IAM v1

### PR DESCRIPTION
**Pull request by dbozzato81**
Included support for GLP IAM v1.
Following 2 new headers have been included to support GLP IAM v1 authZ:

- `X-Role`: GLP Role Name
- `X-Workspaceid`: GLP workspace ID

### CODE REVIEW

File: v1/api/swagger/paths/allocation@storage.yaml [hunk 1]
```diff
@@ -4,8 +4,11 @@ get:
   summary: Get storage allocation
   description: | 
     Returns an array of allocation information for each server instance type. Each instance type's allocation information is further grouped by service type.
-    When using a Metal token, the value in the 'Membership' header determines the scope of the response, i.e., Project or Hoster. However, the default scope is Tenant/Hoster when using the GL IAM token. If the allocation data is needed for a specific project, then the 'Project' header must be present in the request.
+    When using a Metal token, the value in the 'Membership' header determines the scope of the response, i.e., Project or Hoster. However, the default scope is Tenant/Hoster when using the GL IAM token. 
+    If the allocation data is needed for a specific project, then the 'Project' header must be present in the request.
     If the 'siteID' query parameter is present, the information returned is specific to that site ID.
+    If GreenLake Platform IAM issued token is used for authentication, then it is required to pass 
+    'X-Role' header.
```
Comment (mchuang3) [inline v1/api/swagger/paths/allocation@storage.yaml:11]: Same here, X-WorkspaceID needed?

File: v1/api/swagger/paths/projects@{projectId}.yaml [hunk 2]
```diff
@@ -43,22 +45,25 @@ put:
   description: >-
     Updates a project with a matching ID. Only Project 'Name', 'Profile' and 'Limits' can be updated
     with this operation.
-    Note that Hoster or BMaaS Access Owner role is required for this operation.
+    Note that a Hoster, BMaaS Access Owner, BMaaS Tenant Owner or Service Platform Owner role 
+    is required for this operation.
```
Comment (mchuang3) [inline v1/api/swagger/paths/projects@{projectId}.yaml:49]: Let's split this up into the separate environment
```suggestion
    Note that a Hoster or BMaaS Access Owner (under GreenLake Cloud Services), or
    BMaaS Tenant Owner or Service Platform Owner role (under GreenLake Platform) 
    is required for this operation.
```

File: v1/api/swagger/paths/projects@{projectId}.yaml [hunk 3]
```diff
@@ -91,22 +96,25 @@ delete:
   summary: Delete a Project
   description: >-
     Deletes the Project with the matching ID.
-    Note that Hoster or BMaaS Access Owner role is required for this operation.
+    Note that a Hoster, BMaaS Access Owner, BMaaS Tenant Owner or Service Platform Owner role 
+    is required for this operation.
```
Comment (mchuang3) [inline v1/api/swagger/paths/projects@{projectId}.yaml:100]: Same here
```suggestion
    Note that a Hoster or BMaaS Access Owner (under GreenLake Cloud Services), or
    BMaaS Tenant Owner or Service Platform Owner role (under GreenLake Platform) 
    is required for this operation.
```

File: v1/api/swagger/paths/services.yaml [hunk 4]
```diff
@@ -5,11 +5,15 @@ get:
   description: >-
     Returns an array of all OS service images objects that have been created.
     If GreenLake IAM issued token is used for authentication,
```
Comment (mchuang3) [inline v1/api/swagger/paths/services.yaml:7]: Maybe do a global grep for "GreenLake IAM" string & make sure they are all replaced with "GreenLake Cloud Services IAM".

```suggestion
    If GreenLake Cloud Services IAM issued token is used for authentication, 
```

File: v1/api/swagger/paths/projects-info.yaml [hunk 5]
```diff
@@ -12,19 +12,20 @@ get:
 
     When GreenLake IAM issued token is used for authentication, it is required to
```
Comment (mchuang3) [inline v1/api/swagger/paths/projects-info.yaml:13]:
```suggestion
    When GreenLake Cloud Services IAM issued token is used for authentication, it is required to 
```

File: v1/api/swagger/paths/services@{serviceId}.yaml [hunk 6]
```diff
@@ -34,6 +38,8 @@ put:
   description: >-
     Updates an OS service image with a matching ID.
     Note that Hoster or BMaaS Access Owner role is required for this operation.
```
Comment (mchuang3) [inline v1/api/swagger/paths/services@{serviceId}.yaml:40]: This is GLCS only restriction. Still TODO on GLP side, but no need to comment on that.
```suggestion
    Note that Hoster or BMaaS Access Owner role is required for this operation under GreenLake Cloud Service.
```

File: v1/api/swagger/paths/services@{serviceId}.yaml [hunk 7]
```diff
@@ -79,6 +87,8 @@ delete:
   description: >-
     Deletes the OS service image with the matching ID.
     Note that Hoster or BMaaS Access Owner role is required for this operation.
```
Comment (mchuang3) [inline v1/api/swagger/paths/services@{serviceId}.yaml:89]:
```suggestion
    Note that Hoster or BMaaS Access Owner role is required for this operation under GreenLake Cloud Service.
```

File: README.md [hunk 8]
```diff
@@ -8,7 +8,7 @@ Navigate to version-specific API Client directories (\<version\>/pkg/client) to
...
 
-The APIs use bearer authentication (also called token authentication) that requires the user to provide a bearer token in the authorization header. Two token methods are supported, the HPE GreenLake IAM token and Metal classic tokens.
+The APIs use bearer authentication (also called token authentication) that requires the user to provide a bearer token in the authorization header. Three token methods are supported, the HPE GreenLake Cloud Services IAM token, HPE Greenlake Platform IAM token and Metal classic tokens.
```
Comment (anurag-shrivastav) [inline README.md:11]:
```suggestion
The APIs use bearer authentication (also called token authentication) that requires the user to provide a bearer token in the authorization header. Three token methods are supported, the HPE GreenLake Cloud Services IAM token, HPE GreenLake Platform IAM token and Metal classic tokens.
```

File: v1/api/swagger/components/parameters/WorkspaceID.yaml [hunk 9]
```diff
@@ -0,0 +1,6 @@
+name: X-Workspaceid
```
Comment (anurag-shrivastav) [inline v1/api/swagger/components/parameters/WorkspaceID.yaml:1]: X-Workspace**ID** for readability?

**Review by mchuang3 — APPROVED** [review]

**Pull request merged**